### PR TITLE
[SYCLomatic][CMake] Enable target_link_options,target_link_directories

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Ahmed, Daiyaan <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/test/dpct/cmake_migration/case_038/input.cmake [hunk 1]
```diff
@@ -0,0 +1,3 @@
+target_link_directories(CUDA::toolkit INTERFACE "${CUDAToolkit_LIBRARY_DIR}")
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_038/input.cmake]: Please also test test cases: 
                                     | (a empty)                                                                         | (a emtpy)
target_link_directories( CUDA::toolkit INTERFACE "${CUDAToolkit_LIBRARY_DIR}" )

Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_038/input.cmake]: do we need process CUDA::toolkit?

File: clang/test/dpct/cmake_migration/case_038/expected.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+target_link_directories(CUDA::toolkit INTERFACE "${ONEAPI_LIB_DIR}")
+
+target_link_options(quda PUBLIC )
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_038/expected.txt]: is the expected.txt runnable with CMake?

Comment (daiyaan-ahmed6) [reply clang/test/dpct/cmake_migration/case_038/expected.txt]: Yes

File: clang/test/dpct/cmake_migration/case_038/input.cmake (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/test/dpct/cmake_migration/case_038/input.cmake]: No, it is just a target name, could be anything

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by TejaX-Alaghari — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**